### PR TITLE
Update Search module and mock data

### PR DESCRIPTION
**Pull request by mrblippy**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included

##### Description of change

https://github.com/openequella/openEQUELLA/issues/1306

Update Search module and mock data to reflect additional attachment properties (mimeType, isPlaceholderThumb) being added to search2 rest endpoint.

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by mrblippy** [comment]
> Looks good. I guess once you add the next bit though we'll see you'll need some better mock data.

Yep, correct. Planning on expanding out the mock data (to test more cases) in the next part.

**Comment by mrblippy** [comment]
SearchModule.test.ts is currently failing. So I need to fix that now

**Comment by mrblippy** [comment]
> SearchModule.test.ts is currently failing. So I need to fix that now

getSearchResult.js and searchresult_mock_data.ts contain some very similar code. We should probably look to combine them in a future commit

File: oeq-ts-rest-api/src/Search.ts [hunk 1]
```diff
@@ -121,6 +121,14 @@ export interface Attachment {
    * True if an attachment can be previewed.
    */
   preview: boolean;
+  /**
+   * MimeType of file based attachment
+   */
+  mimeType?: string;
+  /**
+   * True if an file attachment has a generated thumbnail in filestore
```
Comment (edalex-ian) [inline oeq-ts-rest-api/src/Search.ts:129]:
```suggestion
   * True if a file attachment has a generated thumbnail in filestore
```

**Pull request merged**